# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1781214624 -7200
# Parent  7df36ac725ce4c1f51f7453888c678b6755de824
Feature: Account colours, part 5: Fix Thunderbird vandalism of displaying useless blue bar when the account has no colour.

Alternatively, the entire
https://hg-edge.mozilla.org/comm-central/rev/b7634da2016fe5b239473b2ac766bbaade07c752
since it offers minimal value.

diff --git a/mail/base/content/widgets/folder-tree-row.mjs b/mail/base/content/widgets/folder-tree-row.mjs
--- a/mail/base/content/widgets/folder-tree-row.mjs
+++ b/mail/base/content/widgets/folder-tree-row.mjs
@@ -263,8 +263,12 @@ class FolderTreeRow extends HTMLLIElemen
     this.accountIndicator.style.setProperty(
       "--account-color",
       `var(--server-${CSS.escape(folder.server.key)}-color)`
     );
+    this.accountIndicator.style.setProperty(
+      "--account-display",
+      `var(--server-${CSS.escape(folder.server.key)}-display)`
+    );
   }
 
   /**
    * Set some properties based on the server for this row.
diff --git a/mail/base/content/widgets/treeview/thread-card.mjs b/mail/base/content/widgets/treeview/thread-card.mjs
--- a/mail/base/content/widgets/treeview/thread-card.mjs
+++ b/mail/base/content/widgets/treeview/thread-card.mjs
@@ -115,8 +115,12 @@ class ThreadCard extends TreeViewTableRo
     this.accountIndicator.style.setProperty(
       "--account-color",
       `var(--server-${CSS.escape(data.serverKey)}-color)`
     );
+    this.accountIndicator.style.setProperty(
+      "--account-display",
+      `var(--server-${CSS.escape(data.serverKey)}-display)`
+    );
     this.accountIndicator.title = data.account;
 
     this.dataset.properties = properties.value.trim();
 
diff --git a/mail/base/content/widgets/treeview/thread-row.mjs b/mail/base/content/widgets/treeview/thread-row.mjs
--- a/mail/base/content/widgets/treeview/thread-row.mjs
+++ b/mail/base/content/widgets/treeview/thread-row.mjs
@@ -302,8 +302,12 @@ class ThreadRow extends TreeViewTableRow
           cell.style.setProperty(
             "--account-color",
             `var(--server-${CSS.escape(cellTexts[textColumns.indexOf("serverKeyCol")])}-color)`
           );
+          cell.style.setProperty(
+            "--account-display",
+            `var(--server-${CSS.escape(cellTexts[textColumns.indexOf("serverKeyCol")])}-display)`
+          );
         }
         if (!column.custom) {
           document.l10n.setAttributes(cell, column.l10n.cell, {
             title: cellTexts[textIndex],
diff --git a/mail/modules/AccountColorUtils.sys.mjs b/mail/modules/AccountColorUtils.sys.mjs
--- a/mail/modules/AccountColorUtils.sys.mjs
+++ b/mail/modules/AccountColorUtils.sys.mjs
@@ -80,17 +80,20 @@ export const AccountColorUtils = {
         continue;
       }
 
       const serverProperty = `--server-${CSS.escape(server.key)}-color`;
+      const serverProperty2 = `--server-${CSS.escape(server.key)}-display`;
       const color = lazy.FolderTreeProperties.getColor(server.rootFolder.URI);
       // Try to clear the property if no color is available in order to account
       // for custom colors that have been reset to default.
       if (!color) {
         style.removeProperty(serverProperty);
+        style.setProperty(serverProperty2, "none");
         continue;
       }
 
       style.setProperty(serverProperty, color);
+      style.setProperty(serverProperty2, "inline-flex");
     }
   },
 
   /**
diff --git a/mail/themes/shared/mail/about3Pane.css b/mail/themes/shared/mail/about3Pane.css
--- a/mail/themes/shared/mail/about3Pane.css
+++ b/mail/themes/shared/mail/about3Pane.css
@@ -696,8 +696,9 @@ conversation-view:not([hidden]) {
 
 .account-indicator {
   display: none;
   box-sizing: border-box;
+  /* Removing var(--folder-color-inbox) would make the indicator transparent, but it would still occupy width */
   background-color: var(--account-color, var(--folder-color-inbox));
   border-radius: 3px;
   inline-size: 3px;
   block-size: 1rem;
@@ -743,6 +744,6 @@ conversation-view:not([hidden]) {
     [data-folder-type="trash"]
   )
   > .container
   > .account-indicator {
-  display: inline-flex;
+  display: var(--account-display);
 }
diff --git a/mail/themes/shared/mail/threadCard.css b/mail/themes/shared/mail/threadCard.css
--- a/mail/themes/shared/mail/threadCard.css
+++ b/mail/themes/shared/mail/threadCard.css
@@ -365,9 +365,9 @@
           gap: 6px;
         }
 
         .is-unified & .account-indicator {
-          display: inline-flex;
+          display: var(--account-display);
         }
 
         & :is(.sender, .subject) {
           flex: 1 1 0;
diff --git a/mail/themes/shared/mail/threadPane.css b/mail/themes/shared/mail/threadPane.css
--- a/mail/themes/shared/mail/threadPane.css
+++ b/mail/themes/shared/mail/threadPane.css
@@ -746,6 +746,6 @@ tr[data-properties~="outgoing"].selected
 
 /* Account indicator */
 
 .is-unified tr:not([data-properties~="dummy"]) .accountcol-column::before {
-  display: inline-flex;
+  display: var(--account-display);
 }
